### PR TITLE
Feature

### PR DESCRIPTION
**Pull request by certilremy**
- [x] Include the bcrypt-ruby gem in your Gemfile
- [x] Add the #has_secure_password method to your User file.
- [x] create a sessions_controller.rb and the corresponding routes. Make “sign-in” links in the 
       layout as necessary.
- [x] Set your current user whenever a user signs in.
- [x] Build sign out functionality in your SessionsController#delete
- [x] make sure only signed-in users can see the author of each post.
- [x] only users who are signed in can create a post.

### CODE REVIEW

File: app/helpers/sessions_helper.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  end
+
+  def logout
+    session.clear
+    redirect_to root_path
+  end
+
+  def current_user
+    @current_user ||= User.find(session[:user_id]) if session[:user_id]
+  end
```
Comment (maelfosso) [inline app/helpers/sessions_helper.rb]: Change it to use `cookies.permanent`

File: app/helpers/sessions_helper.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
+module SessionsHelper
+  def login_user(user)
+    session[:user_id] = user.id
+  end
+
+  def logout
+    session.clear
```
Comment (maelfosso) [inline app/helpers/sessions_helper.rb]: cookies, please.

**Review by maelfosso — CHANGES_REQUESTED** [review]
# Changes required :red_circle: 

Hello, 

Thank you for that simplest result. The project is working.
However, there are some requirements that are not met

1. As indicated [here](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/bcrypt_requirement.png), The application uses the bcrypt gem for authentication but you didn't use it in your `User` model.

2. It was asked  A `:remember_token` column to be added to the model through a migration (take a look in that [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/remember_token_requirement.png)), but there is no `remember_token`

3. a #before_create callback that generates a :remember_token which is asked [here](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/before_create_requirement.png) it 's not present

4. Please, forgot the usage of `session` in this project. The authentication should be handled only and only using `cookies`. As mentioned [here](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/sign_in_method_requirement.png), your `log_in / sign in` method should hash the remember token into `cookies.permanent`

The remember_me token the project describes consists of storing a hash in a cookie and using it to determine in a user is logged in. Do not confuse it with the Remember me [ ] checkbox that many websites show. That is a more complex solution, which is not required here

5. As specified [here](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/current_user_requirement.png), It uses the `#current_user=(user)` method to set user

5. Read the comments below.

Kindly make the changes and submit a new code review request.
Thanks

Mael FOSSO

**Comment by certilremy** [comment]
> 1. As indicated [here](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/bcrypt_requirement.png), The application uses the bcrypt gem for authentication but you didn't use it in your `User` model.

Rep. Thank you for your feedback, We will do the change about cookies. One last thing you said that we do not use bcrypt for our authentication. We didn't use it as you expected but we use it. Look at the session controller and the create method. if you want to verify,  fire up the console create a new user . You'll see your password is encrypted. There are various ways u can use bcrypt to encrypt password. We hate plagiarism this is why we come up with our own implementation method. But if you want us to copy code from the tutorial and past it to the project it fine We will remove our implementation.

**Review by Torres-ssf — APPROVED** [review]
### Project Approved :clap: :tada: 
Hi team :smile: 
Good work implementing the requested changes :1st_place_medal: 
**I'm approving this PR.** :smiley: 
Good luck with the next project :+1: 

Happy coding :computer: 
**Sergio Torres**

**Pull request merged**